src/features/metadata-based-view/MetadataBasedItemList.js

```diff
@@ -101,18 +111,50 @@ class MetadataBasedItemList extends React.Component<Props, State> {
         onItemClick(itemWithPreviewPermission);
     }
 
-    getGridCellData(columnIndex: number, rowIndex: number): GridCellData {
+    handleEditIconClick(columnIndex: number, rowIndex: number): void {
```

Comment:
Should this be an arrow function as well?

Reply:
No. This works because on filename.onClick is an arrow function.
```
filename.onClick = () => this.handleEditIconClick(columnIndex, rowIndex);
```
Reference to `this` is maintained when the onClick handler is called.